nsv13/code/game/objects/items/bloody_eye.dm

```diff
@@ -0,0 +1,196 @@
+
+/datum/reagent/drug/bloody_eye/on_mob_end_metabolize(mob/living/L)
+	REMOVE_TRAIT(L,TRAIT_MONKEYLIKE,type)
+	REMOVE_TRAIT(L,TRAIT_NOCRITDAMAGE,type)
+	REMOVE_TRAIT(L,TRAIT_NOSOFTCRIT,type)
+	REMOVE_TRAIT(L,TRAIT_NOHARDCRIT,type)
+	REMOVE_TRAIT(L,TRAIT_NOBLOCK,type)
+	REMOVE_TRAIT(L,TRAIT_STUNIMMUNE,type)
+	REMOVE_TRAIT(L,TRAIT_SLEEPIMMUNE,type)
+	REMOVE_TRAIT(L,TRAIT_IGNOREDAMAGESLOWDOWN,type)
+	REMOVE_TRAIT(L,TRAIT_NOSTAMCRIT,type)
+	REMOVE_TRAIT(L,TRAIT_NOLIMBDISABLE,type)
+	L.remove_status_effect(/datum/status_effect/bloody_eye)
+	L.remove_movespeed_modifier(type)
+	..()
+
+/datum/reagent/drug/bloody_eye/on_mob_life(mob/living/carbon/M)
+	var/high_message = pick("You see red.", "You feel like ripping out someone's throat.", "You feel like nothing could ever kill you.")
+	if(prob(5))
+		to_chat(M, "<span class='notice'>[high_message]</span>")
+	M.AdjustStun(-40, FALSE)
+	M.AdjustKnockdown(-40, FALSE)
+	M.AdjustUnconscious(-40, FALSE)
+	M.AdjustParalyzed(-40, FALSE)
+	M.AdjustImmobilized(-40, FALSE)
+	M.adjustStaminaLoss(-30, 0)
+	M.Jitter(2)
+	M.adjustOrganLoss(ORGAN_SLOT_BRAIN, 0.25)
+	if(prob(5))
+		M.emote(pick("scream", "laugh"))
+	..()
+	. = 1
+
+/datum/reagent/drug/bloody_eye/overdose_start(mob/living/M)
+	to_chat(M, "<span class='userdanger'>KILLKILLKILLKILLKILLKILLKILLKILL!</span>")
+	SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "[type]_overdose", /datum/mood_event/overdose, name)
+
+/datum/reagent/drug/bloody_eye/overdose_process(mob/living/M)
+	if((M.mobility_flags & MOBILITY_MOVE) && !ismovableatom(M.loc))
+		for(var/i in 1 to 4)
+			step(M, pick(GLOB.cardinals))
+	if(prob(20))
+		M.emote("scream")
+	if(prob(33))
+		M.visible_message("<span class='danger'>[M]'s eyes are blood red!</span>")
+		M.drop_all_held_items()
+	..()
+	M.adjustToxLoss(0.25, 0)
+	M.adjustOrganLoss(ORGAN_SLOT_BRAIN, pick(0.1, 0.15, 0.2, 0.25))
+	. = 1
+
+/datum/reagent/drug/bloody_eye/addiction_act_stage1(mob/living/M)
+	M.Jitter(20)
+	if(prob(20))
+		M.emote(pick("scream","laugh"))
+	..()
+
+/datum/reagent/drug/bloody_eye/addiction_act_stage2(mob/living/M)
+	M.Jitter(50)
+	M.Dizzy(5)
+	if(prob(30))
+		M.emote(pick("scream","gasp","laugh"))
+	..()
+
+/datum/reagent/drug/bloody_eye/addiction_act_stage3(mob/living/M)
+	if((M.mobility_flags & MOBILITY_MOVE) && !ismovableatom(M.loc))
+		for(var/i = 0, i < 4, i++)
+			step(M, pick(GLOB.cardinals))
+	M.Jitter(100)
+	M.Dizzy(10)
+	if(prob(40))
+		M.emote(pick("scream","gasp"))
+	..()
+
+/datum/reagent/drug/bloody_eye/addiction_act_stage4(mob/living/carbon/human/M)
+	if((M.mobility_flags & MOBILITY_MOVE) && !ismovableatom(M.loc))
+		for(var/i = 0, i < 8, i++)
+			step(M, pick(GLOB.cardinals))
+	M.Jitter(150)
+	M.Dizzy(15)
+	M.adjustToxLoss(0.25, 0)
+	if(prob(50))
+		M.emote(pick("scream"))
+	..()
+	. = 1
+
+/obj/item/reagent_containers/hypospray/bloody_eye/attack_paw(mob/user)
+	return attack_hand(user)
+
+/obj/item/reagent_containers/hypospray/bloody_eye/attack(mob/living/M, mob/user)
```

Comment:
99% inherited from parent minus flavour text.
However, this is not checking if eyes are covered or not.

Either rewrite or use parent handling.

Reply:
Changed check to check if eyes are covered or not.